### PR TITLE
ensure git-lfs is installed to avoid confusing errors

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- This is to avoid confusing errors when `git-lfs` is not installed, e.g., 
```
--> src/daemon/bundle.rs:16:5
  |
16 |   const_assert_eq!(ACTOR_BUNDLES_CAR_ZST.len(), 2438387);
  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
```
even on a new git clone.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by elmattic — APPROVED** [review]

**Comment by hanabi1224** [comment]
cargo install forest-filecoin should not require git-lfs

**Review by jdjaustin — APPROVED** [review]

**Review by aatifsyed — CHANGES_REQUESTED** [review]
+1 for `cargo install` not requiring `git lfs`

**Comment by LesnyRumcajs** [comment]
> cargo install forest-filecoin should not require git-lfs
> +1 for cargo install not requiring git lfs

@hanabi1224 @aatifsyed And how would you achieve this? I encourage you to remove `git-lfs` from your system temporarily, git clone a new repo and run `make install`. This is what triggered this change.

**Comment by aatifsyed** [comment]
We have two end-user install pathways:
- `git clone ... && make install`
- `cargo install forest-filecoin`

If a user performed the latter, they would erroneously require git-lfs, even though git isn't in the loop - the bundle is in the cargo package. 

I'd need to know more about the code that includes the actor bundles, but I suspect it's a case of checking that they're there already or something. 

(Note that I don't think we should support `cargo install` in general - it's causing a lot of warts like this >.<)

**Comment by aatifsyed** [comment]
Having read the above, maybe my comment is a bit of a nit "we don't require exactly the minimal number of installation dependencies"

**Comment by LesnyRumcajs** [comment]
> We have two end-user install pathways:
> 
> * `git clone ... && make install`
> * `cargo install forest-filecoin`
> 
> If a user performed the latter, they would erroneously require git-lfs, even though git isn't in the loop - the bundle is in the cargo package.

I understand now, thanks. It's a valid concern. I don't quite know how to get around this without nastily exposing Forest internals to the build script.

**Comment by hanabi1224** [comment]
> I understand now, thanks. It's a valid concern. I don't quite know how to get around this without nastily exposing Forest internals to the build script.

@LesnyRumcajs How about checking bundle size before `git-lfs`, in case bundle size is invalid, `cargo publish` would fail, and the published crate should always have correct bundle size thus `git-lfs` check would be bypassed

**Comment by LesnyRumcajs** [comment]
> > I understand now, thanks. It's a valid concern. I don't quite know how to get around this without nastily exposing Forest internals to the build script.
> 
> @LesnyRumcajs How about checking bundle size before `git-lfs`, in case bundle size is invalid, `cargo publish` would fail, and the published crate should always have correct bundle size thus `git-lfs` check would be bypassed

I don't think I get it. We are already checking the bundle size, and ending up with the cryptic message in the PR description, no?

**Comment by hanabi1224** [comment]
> I don't think I get it. We are already checking the bundle size, and ending up with the cryptic message in the PR description, no?

@LesnyRumcajs If we move the bundle size check to the build script, we don't have to use `const_assert_eq!`, a normal `assert_eq!` with error message would just work

**Review by elmattic — APPROVED** [review]

File: build.rs [hunk 1]
```diff
@@ -9,9 +10,46 @@ const PROTO_DIR: &str = "proto";
...
+    if is_bundle_size_correct()? {
+        Ok(())
+    } else {
+        bail!(
+            "actor bundle size is incorrect. \
+            Please run `git lfs pull` and try again. If this problem persists, \
+            please open an issue."
+        )
+    }
+}
```
[the remark this answers is not in the document]
Comment (aatifsyed) [reply build.rs:51]: should probably even be `fn check_bundle` - we don't capture any state

[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply build.rs:51]: So now I'm reviewing your code in my code review? That's proper reviewoception

**Pull request merged**